src/include/platform/NetworkCommissioning.h

```diff
@@ -355,21 +361,23 @@ class WiFiDriver : public Internal::WirelessDriver
      *  @brief Provide all the frequency bands supported by the Wi-Fi interface
      *
      *  The default implementation returns the 2.4 GHz band support.
-     *  Note: WiFi platforms should implement this function in their WiFiDriver to provide their complete device capabilities
+     *  Note: WiFi platforms should implement this function in their WiFiDriver to provide their complete device capabilities.
      *
-     *  @param bands Reference to a span that is used to return the supported bands.
-     *               The span is initialized with a fixed-size buffer.
-     *               The implementation is expected to resize the span to match the number of returned results.
+     *  The returned bit mask has values of WiFiBandEnum packed into the bits. For example:
      *
-     *  @return CHIP_NO_ERROR on success or a CHIP_ERROR on failure.
+     *    - Bit 0 = (WiFiBandEnum::k2g4 == 0) --> (1 << 0) == (1 << WiFiBandEnum::k2g4)
+     *    - Bit 2 = (WiFiBandEnum::k5g == 2) --> (1 << 2) == (1 << WiFiBandEnum::k5g)
+     *    - If both 2.4G and 5G are supported --> ((1 << k2g4) || (1 << k5g)) == (1 || 4) == 5
+     *
+     *  On error, return 0 (no bands supported). This should never happen... Note that
+     *  certification tests will REQUIRE at least one bit set in the set.
+     *
+     *  @return a bitmask of supported Wi-Fi bands where each bit is associated with a WiFiBandEnum value.
      */
-    virtual CHIP_ERROR GetSupportedWiFiBands(Span<WiFiBand> & bands)
+    virtual uint32_t GetSupportedWiFiBands()
```

Comment:
nit: consider naming this GetSupportedWiFiBandsMask.

Could this be a BitMask<WiFiBandEnum, uint32_t> ? That seems to force a bit more type safety while still being able to use Raw() and return simple values without list unpacking.

Reply:
Will do as a follow-up.